billing/src/main/java/com/appcoins/wallet/billing/carrierbilling/CarrierBillingRepository.kt

```diff
@@ -0,0 +1,69 @@
+package com.appcoins.wallet.billing.carrierbilling
+
+import com.appcoins.wallet.billing.carrierbilling.request.CarrierTransactionBody
+import com.appcoins.wallet.billing.carrierbilling.response.CarrierCreateTransactionResponse
+import com.appcoins.wallet.billing.common.response.TransactionResponse
+import io.reactivex.Observable
+import io.reactivex.Single
+import okhttp3.ResponseBody
+import retrofit2.Response
+import retrofit2.http.*
+
+class CarrierBillingRepository(private val api: CarrierBillingApi,
+                               private val mapper: CarrierResponseMapper,
+                               packageName: String) {
+
+  companion object {
+    private const val METHOD = "onebip"
+    const val GATEWAY_NAME = "dimoco"
+  }
+
+  private val RETURN_URL = "http://${packageName}/return/carrier_billing"
+
+  fun makePayment(walletAddress: String, walletSignature: String,
+                  phoneNumber: String, packageName: String, origin: String?, sku: String?,
+                  reference: String?, transactionType: String, currency: String,
+                  value: String, developerWallet: String, oemWallet: String, storeWallet: String,
+                  userWallet: String): Single<CarrierPaymentModel> {
```

Comment:
I believe the wallets can be null, didn't it crash on Buy oil or buy anti freeze?